aws/resource_aws_launch_template_test.go

```diff
@@ -0,0 +1,228 @@
+					resource.TestCheckResourceAttr(resName, "default_version", "1"),
+					resource.TestCheckResourceAttr(resName, "latest_version", "1"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSLaunchTemplate_data(t *testing.T) {
+	var template ec2.LaunchTemplate
+	resName := "aws_launch_template.foo"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSLaunchTemplateDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSLaunchTemplateConfig_data,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSLaunchTemplateExists(resName, &template),
+					resource.TestCheckResourceAttr(resName, "block_device_mappings.#", "1"),
+					resource.TestCheckResourceAttr(resName, "credit_specification.#", "1"),
+					resource.TestCheckResourceAttrSet(resName, "disable_api_termination"),
+					resource.TestCheckResourceAttr(resName, "elastic_gpu_specifications.#", "1"),
+					resource.TestCheckResourceAttr(resName, "iam_instance_profile.#", "1"),
+					resource.TestCheckResourceAttrSet(resName, "image_id"),
+					resource.TestCheckResourceAttrSet(resName, "instance_initiated_shutdown_behavior"),
+					resource.TestCheckResourceAttr(resName, "instance_market_options.#", "1"),
+					resource.TestCheckResourceAttrSet(resName, "instance_type"),
+					resource.TestCheckResourceAttrSet(resName, "kernel_id"),
+					resource.TestCheckResourceAttrSet(resName, "key_name"),
+					resource.TestCheckResourceAttrSet(resName, "monitoring"),
+					resource.TestCheckResourceAttr(resName, "network_interfaces.#", "1"),
+					resource.TestCheckResourceAttr(resName, "placement.#", "1"),
+					resource.TestCheckResourceAttrSet(resName, "ram_disk_id"),
+					resource.TestCheckResourceAttr(resName, "vpc_security_group_ids.#", "1"),
+					resource.TestCheckResourceAttr(resName, "tag_specifications.#", "1"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSLaunchTemplate_update(t *testing.T) {
+	var template ec2.LaunchTemplate
+	resName := "aws_launch_template.foo"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSLaunchTemplateDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSLaunchTemplateConfig_basic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSLaunchTemplateExists(resName, &template),
+					resource.TestCheckResourceAttr(resName, "default_version", "1"),
+					resource.TestCheckResourceAttr(resName, "latest_version", "1"),
+				),
+			},
+			{
+				Config: testAccAWSLaunchTemplateConfig_data,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSLaunchTemplateExists(resName, &template),
+					resource.TestCheckResourceAttr(resName, "default_version", "1"),
+					resource.TestCheckResourceAttr(resName, "latest_version", "2"),
+					resource.TestCheckResourceAttrSet(resName, "image_id"),
+				),
+			},
```

Comment:
Let's add a `TestStep` here that imports the resource and ensures the Terraform state matches what is expected 👍 

```
{
  ResourceName: resName,
  ImportState: true,
  ImportStateVerify: true,
},
```